examples/knative/powershell/kn-ps-harbor-slack/README.md

```diff
@@ -0,0 +1,112 @@
+
+```console
+Testing Function ...
+See docker container console for output
+
+# Output from docker container console
+Detected change to subject-123 ...
+Sending Webhook payload to Slack ...
+StatusCode        : 200                                                                                                                                                                                                            StatusDescription : OK                                                                                                                                                                                                             Content           : ok
+RawContent        : HTTP/1.1 200 OK
+                    Date: Tue, 16 Feb 2021 21:56:16 GMT
+                    Server: Apache
+                    Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
+                    x-slack-backend: r
+                    referrer-policy: no-referrer
+                    Vary: Accept-…
+Headers           : {[Date, System.String[]], [Server, System.String[]], [Strict-Transport-Security, System.String[]], [x-slack-backend, System.String[]]…}
+Images            : {}
+InputFields       : {}
+Links             : {}
+RawContentLength  : 2
+RelationLink      : {}
+```
+
+# Step 3 - Deploy
+
+> **Note:** The following steps assume a working Knative environment using the
+`default` Rabbit `broker`. The Knative `service` and `trigger` will be installed in the
+`vmware-functions` Kubernetes namespace, assuming that the `broker` is also available there.
+>
+> **Note:** Also, in order to receive incoming Harbor events and to ultimately invoke the Harbor-Slack-Function, it's necessary to have the [kn-go-harbor-webhook function example](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/development/examples/knative/go/kn-go-harbor-webhook) running properly first.
+
+Push your container image to an accessible registry such as Docker once you're done developing and testing your function logic.
+
+```console
+docker push <docker-username>/kn-ps-harbor-slack:${TAG}
+```
+
+Update the `slack_secret.json` file with your Slack webhook configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `SLACK_SECRET`.
+
+```console
+# create secret
+
+kubectl -n vmware-functions create secret generic harbor-slack-secret --from-file=SLACK_SECRET=slack_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret slack-secret app=veba-ui
+```
+
+Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `com.vmware.harbor.push_artifact.v0` Harbor Event. If you wish to change this, update the `type` field within `function.yaml` to the desired event type.
```

Comment:
nit: Here some help would be nice, e.g. a link to Harbor event notifications and/or how to inspect them, e.g. Sockeye

Reply:
Added:
A list of supported notification events is available on the official Harbor documentation under [Configure Webhook Notifications](https://goharbor.io/docs/2.5.0/working-with-projects/project-configuration/configure-webhooks/). Furthermore, use the VEBA Event viewer endpoint (`https://<VEBA-FQDN>/events`) to display all incoming events.

Reply:
Done